app/src/main/java/com/scanner/cardreader/CameraActivity.java

```diff
@@ -42,68 +40,75 @@ protected void onCreate(Bundle savedInstanceState) {
         // underlying surface is created and destroyed.
 
         surfaceHolder.addCallback(this);
+        simInfo = (TextView) findViewById(R.id.simInfo);
+        simInfo.setText(MainActivity.SIM);
 
 // deprecated setting, but required on Android versions prior to 3.0
 
         surfaceHolder.setType(SurfaceHolder.SURFACE_TYPE_PUSH_BUFFERS);
 
-        ImageButton button  = (ImageButton) findViewById(R.id.imageButton);
+        ImageButton button = (ImageButton) findViewById(R.id.imageButton);
         //        Check if there is a camera on the device
 
-        if(!getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA)){
-            Toast.makeText(this,"No camera present on this device", Toast.LENGTH_LONG).show();
-        }
-        else{
+        if (!getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
+            Toast.makeText(this, "No camera present on this device", Toast.LENGTH_LONG).show();
+        } else {
             cameraId = findRearFacingCamera();
-            if(cameraId < 0){
+            if (cameraId < 0) {
                 Toast.makeText(this, "Rear facing camera not found", Toast.LENGTH_LONG).show();
-            }
-            else{
+            } else {
                 camera = Camera.open(cameraId);
             }
         }
 
+
         jpegCallBack = new Camera.PictureCallback() {
             @Override
             public void onPictureTaken(byte[] bytes, Camera camera) {
-                try{
-                    if(bytes != null){
+                try {
+                    if (bytes != null) {
+                        Intent i = new Intent(getApplicationContext(), CropActivity.class);
                         Bitmap bitmap = BitmapFactory.decodeByteArray(bytes, 0, bytes.length);
-                        Log.d(tag,Integer.toString(bitmap.getByteCount()));
+                        setBitmapImage(bitmap);
+                        startActivity(i);
+                        camera.stopPreview();
+                        camera.release();
+                        camera.setPreviewCallback(null);
+                        camera = null;
 
                     }
-                }
-                catch(Exception e){
+                } catch (Exception e) {
                     System.out.println(e);
                 }
 
 
             }
         };
-        shutterCallback = new Camera.ShutterCallback() {
-            @Override
-            public void onShutter() {
-                MediaActionSound sound = new MediaActionSound();
-                sound.play(MediaActionSound.SHUTTER_CLICK);
-
-            }
-        };
-
 
 
 
 //Take a picture on button click
+
+        assert button!=null;
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                camera.takePicture(shutterCallback, null,
+                camera.takePicture(null, null,
```

Comment:
This feature was used to play a shutter click sound on image capture, now removed to reduce execution overhead.